pkg/utils/baremetal/cache.go

```diff
@@ -0,0 +1,46 @@
+package baremetal
+
+import (
+	"context"
+	"time"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/watch"
+	"k8s.io/client-go/dynamic"
+)
+
+// BmhCacheListerWatcher is an object that wraps the listing and wrapping
+// functionality for baremetal host resources.
+type BmhCacheListerWatcher struct {
+	Resource dynamic.ResourceInterface
+}
+
+// List returns a list of baremetal hosts as dynamic objects.
+func (bc BmhCacheListerWatcher) List(options metav1.ListOptions) (runtime.Object, error) {
+	list, err := bc.Resource.List(context.TODO(), options)
+	if err != nil {
+		if err.Error() == "the server could not find the requested resource" {
+			return &unstructured.UnstructuredList{}, nil
+		}
+	}
+	return list, err
+}
+
+// Watch starts a watch over baremetal hosts.
+func (bc BmhCacheListerWatcher) Watch(options metav1.ListOptions) (watch.Interface, error) {
+	w, err := bc.Resource.Watch(context.TODO(), options)
+	if err != nil {
+		if err.Error() == "the server could not find the requested resource" {
+			// We can't use watch.NewEmptyWatch here because it closes too quickly.
```

Comment:
I'm sceptical of this comment because we previously [established](https://github.com/openshift/installer/pull/8181#discussion_r1548863672) that you accidentally had code that was equivalent to returning `watch.NewEmptyWatch()`, and that was working for you.

Or to put it another way, what's the difference between sleeping for 2s here vs. blocking for 2s when something calls the watcher?

Reply:
I have done a whole bunch of testing and research.  The caller starts a timer before it calls us and if the watch we return stops in under one second, it raises an error.  If the watch closes after that cut off, the caller will retry.  We have two options:

1. Return a fake watch and use a goroutine to call the stop function asynchronously. 
2. Return a `NewEmphyWatch` and introduce a synchronous fake delay of one second to force the caller to retry.

I have done the latter and explained in a comment. Happy to explore other ideas.